akka-typed/src/main/scala/akka/typed/Behavior.scala

```diff
@@ -0,0 +1,242 @@
+/**
+ * Copyright (C) 2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.typed
+
+import scala.concurrent.duration.FiniteDuration
+import scala.reflect.ClassTag
+import akka.actor.OneForOneStrategy
+import scala.annotation.tailrec
+import scala.collection.immutable
+
+/**
+ * The behavior of an actor defines how it reacts to the messages that it
+ * receives. The message may either be of the type that the Actor declares
+ * and which is part of the [[ActorRef]] signature, or it may be a system
+ * [[Signal]] that expresses a lifecycle event of either this actor or one of
+ * its child actors.
+ *
+ * Behaviors can be formulated in a number of different ways, either by
+ * creating a derived class or by employing factory methods like
+ * [[Behavior.Simple]], [[Behavior.Static]], [[Behavior.Full]] etc.
+ */
+abstract class Behavior[T] { self ⇒
```

Comment:
In the code below `this` refers to the anonymous Behavior just created. An alternative to providing the outer Behavior’s reference with the `self` name would typically be to say `Behavior.this` instead of using self-type syntax, but in this case that might not even work. In any case, the need for this went away in my branch since I moved `.widen` out into the ScalaDSL where it belongs, will push soon).